components/Cards/examples/EditReorderCollectionWithImagesViewController.swift

```diff
@@ -0,0 +1,256 @@
+    }
+    #else
+    preiOS11Constraints()
+    #endif
+
+    self.updateTitle()
+  }
+
+  func preiOS11Constraints() {
+    self.view.addConstraints(NSLayoutConstraint.constraints(withVisualFormat: "H:|[view]|",
+                                                            options: [],
+                                                            metrics: nil,
+                                                            views: ["view": collectionView]));
+    self.view.addConstraints(NSLayoutConstraint.constraints(withVisualFormat: "V:|[view]|",
+                                                            options: [],
+                                                            metrics: nil,
+                                                            views: ["view": collectionView]));
+  }
+
+  func updateTitle() {
+    if toggle == .edit {
+      navigationItem.rightBarButtonItem?.title = "Reorder"
+      self.title = "Cards (Edit)"
+    } else if toggle == .reorder {
+      navigationItem.rightBarButtonItem?.title = "Edit"
+      self.title = "Cards (Reorder)"
+    }
+  }
+
+  func toggleModes() {
+    if toggle == .edit {
+      toggle = .reorder
+    } else if toggle == .reorder {
+      toggle = .edit
+    }
+    self.updateTitle()
+    collectionView.reloadData()
+  }
+
+  func collectionView(_ collectionView: UICollectionView,
+                      cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+
+    let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "Cell", for: indexPath)
+    guard let cardCell = cell as? CardEditReorderCollectionCell else { return cell }
+
+    MDCCardThemer.applyScheme(cardScheme, toCardCell: cardCell)
+    cardCell.isSelectable = (toggle == .edit)
+
+    let title = dataSource[indexPath.item].1
+    let imageName = dataSource[indexPath.item].0
+    cardCell.configure(title: title, imageName: imageName)
+    cardCell.isSelected = dataSource[indexPath.item].2
+
+    cardCell.isAccessibilityElement = true
+    cardCell.accessibilityLabel = title
+
+    return cardCell
+  }
+
+  func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+    if toggle == .edit {
+      dataSource[indexPath.item].2 = !dataSource[indexPath.item].2
+    }
+  }
+
+  func numberOfSections(in collectionView: UICollectionView) -> Int {
+    return 1
+  }
+
+  func collectionView(_ collectionView: UICollectionView,
+                      numberOfItemsInSection section: Int) -> Int {
+    return dataSource.count
+  }
+
+  func collectionView(_ collectionView: UICollectionView,
+                      moveItemAt sourceIndexPath: IndexPath,
+                      to destinationIndexPath: IndexPath) {
+  }
+
+  func collectionView(_ collectionView: UICollectionView,
+                      layout collectionViewLayout: UICollectionViewLayout,
+                      sizeForItemAt indexPath: IndexPath) -> CGSize {
+    let cardSize = (collectionView.bounds.size.width / 3) - 12
```

Comment:
I'm probably missing something here, but why minus 12? If there are 3 columns and the spacing between items is 8, shouldn't it be minus 16?

Reply:
Not sure what's the meaning of 12. 16 didn't work either.
Anyway, this made more sense to me:
    ``` (collectionView.bounds.size.width - 8 * 4) / 3```
      - First subtract interim spaces & insets, than divide what's left to 3:

Reply:
Sorry, so is the calculation in the code incorrect?